### PR TITLE
Annessja Code Review #2

### PR DESCRIPTION
**Pull request by JoshAnness**
**Analysis of the program:** Cookit! is an all-inclusive meal planning and preparation app that lets you find and create recipes. The app has many of its core functions already in place and a working UI. 

**Was the program available in UC Github on time:** Yes

**Is the program documented/commented well enough for you to understand:** The naming conventions are quite clear. In areas where there could be confusion, comments are available to viewers. 

**Does the program compile:** Yes

**Rationale behind your changes:** Most of my changes were based on the kotlinlang coding conventions [(https://kotlinlang.org/docs/coding-conventions.html)](https://kotlinlang.org/docs/coding-conventions.html). These changes included: indentation, modifiers, string resources, whitespace, and trailing commas. Changes did not change any functionality, just readability and decreasing technical debt. 

**Commits to my project:**
[1](https://github.com/JoshAnness/Checkpoint/commit/3bae831bf410d8a066f2262fcfe16070ce03071a)
[2](https://github.com/JoshAnness/Checkpoint/commit/247a9c10f8dd7c324574a3197dc6095195995f84)
[3](https://github.com/JoshAnness/Checkpoint/commit/8d725722bec0f20e9fe8996cee5ccbbca023cc67)
[4](https://github.com/JoshAnness/Checkpoint/commit/389753415a010aa423f73defdc30e8dc3c6b234c)

**3 Things I learned:**

1. The use of GsonBuilder
2. Mutable Live Data
3. How to use retrofit

### CODE REVIEW

File: app/src/main/java/com/cookit/MainActivity.kt [hunk 1]
```diff
@@ -22,10 +22,11 @@ import org.koin.androidx.viewmodel.ext.android.viewModel
 import androidx.compose.runtime.livedata.observeAsState
 
 class MainActivity : ComponentActivity() {
-    private val viewModel : MainViewModel by viewModel<MainViewModel>()
```
Comment (JoshAnness) [inline app/src/main/java/com/cookit/MainActivity.kt:25]: Horizontal whitespace convention based on [https://kotlinlang.org/docs/coding-conventions.html#horizontal-whitespace](https://kotlinlang.org/docs/coding-conventions.html#horizontal-whitespace)

File: app/src/main/java/com/cookit/MainActivity.kt [hunk 2]
```diff
@@ -75,7 +76,7 @@ class MainActivity : ComponentActivity() {
                     }
                 )
                 {
-                    Text(text = "Search")
+                    Text(text = stringResource(R.string.Search))
```
Comment (JoshAnness) [inline app/src/main/java/com/cookit/MainActivity.kt:79]: Made the string into a stringResource based on [https://developer.android.com/guide/topics/resources/string-resource](https://developer.android.com/guide/topics/resources/string-resource)

File: app/src/main/java/com/cookit/MainActivity.kt [hunk 3]
```diff
@@ -85,7 +86,7 @@ class MainActivity : ComponentActivity() {
                     }
                 )
                 {
-                    Text(text = "Save")
+                    Text(text = stringResource(R.string.Save))
```
Comment (JoshAnness) [inline app/src/main/java/com/cookit/MainActivity.kt:89]: Made the string into a stringResource based on https://developer.android.com/guide/topics/resources/string-resource

File: app/src/main/java/com/cookit/MainActivity.kt [hunk 4]
```diff
@@ -120,54 +121,54 @@ class MainActivity : ComponentActivity() {
...
-            ),
-            onDismissRequest = onDismissRequest
-        ) {
-            list.forEach { text ->
-                DropdownMenuItem(onClick = {
-                    setValue(
-                        TextFieldValue(
-                            text.toString(),
-                            TextRange(text.toString().length)
+    @Composable
```
Comment (JoshAnness) [inline app/src/main/java/com/cookit/MainActivity.kt:124]: Indentation was off, indented once, horizontal whitespace conventions - [https://kotlinlang.org/docs/coding-conventions.html#horizontal-whitespace](https://kotlinlang.org/docs/coding-conventions.html#horizontal-whitespace)

File: app/src/main/java/com/cookit/MainViewModel.kt [hunk 5]
```diff
@@ -13,9 +13,10 @@ import kotlinx.coroutines.launch
  * Used to supply [MutableLiveData] of type [ArrayList] of Recipe to views
  */
 class MainViewModel(var recipeService: IRecipeService = RecipeService()) : ViewModel() {
+
     val recipes: MutableLiveData<ArrayList<Recipe>> = MutableLiveData<ArrayList<Recipe>>()
 
-    fun fetchRecipes() {
+    internal fun fetchRecipes() {
```
Comment (JoshAnness) [inline app/src/main/java/com/cookit/MainViewModel.kt:19]: Added modifier based on conventions - [https://kotlinlang.org/docs/coding-conventions.html#modifiers-order](https://kotlinlang.org/docs/coding-conventions.html#modifiers-order)

File: app/src/main/java/com/cookit/RetrofitClientInstance.kt [hunk 6]
```diff
@@ -8,10 +8,11 @@ import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
 
 object RetrofitClientInstance {
+
     private var retrofit: Retrofit? = null
     private const val BASE_URL = "https://www.themealdb.com/api/json/v1/"
     private val recipeDeserializer = RecipeSerializationService()
-    private val customGSON : Gson = GsonBuilder()
+    private val customGSON: Gson = GsonBuilder()
```
Comment (JoshAnness) [inline app/src/main/java/com/cookit/RetrofitClientInstance.kt:15]: Removed colon space based on whitespace convention - [https://kotlinlang.org/docs/coding-conventions.html#horizontal-whitespace](https://kotlinlang.org/docs/coding-conventions.html#horizontal-whitespace)

File: app/src/main/java/com/cookit/dto/Meal.kt [hunk 7]
```diff
@@ -5,7 +5,7 @@ data class Meal(
     var name: String,
     var mealID: String,
     var description: String,
-    var image: String
+    var image: String,
```
Comment (JoshAnness) [inline app/src/main/java/com/cookit/dto/Meal.kt:8]: Added trailing comma comma based on conventions:
- It makes version-control diffs cleaner – as all the focus is on the changed value.

- It makes it easy to add and reorder elements – there is no need to add or delete the comma if you manipulate elements.

- It simplifies code generation, for example, for object initializers. The last element can also have a comma.

[https://kotlinlang.org/docs/coding-conventions.html#trailing-commas](https://kotlinlang.org/docs/coding-conventions.html#trailing-commas)

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -3,4 +3,6 @@
     <string name="category">Category (ex: Dinner)</string>
     <string name="cuisine">Cuisine (ex: Japanese)</string>
     <string name="recipeName">Recipe Name</string>
+    <string name="Search">Search</string>
```
Comment (JoshAnness) [inline app/src/main/res/values/strings.xml:6]: String values discussed earlier

**Review by JoshAnness — COMMENTED** [review]
Annessja Code Review #2 - Let me know if you have any questions!

File: app/src/main/java/com/cookit/MainActivity.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/cookit/MainActivity.kt:79]: Good idea.  Better for internationalization.

**Comment by discospiff** [comment]
Several good concepts in this review:
- Extract to string resources for internationalization.
- Use accessibility modifiers.
The IDE can format code automatically, and consistently, for all developers on the team.  I'd use this instead of creating a branch, formatting manually, and then merging the branch.

**Pull request merged**